doc/specs/stdlib_stats_distribution_uniform.md

```diff
@@ -0,0 +1,359 @@
+---
+title: stats_distribution
+---
+
+# Statistical Distributions -- Uniform Distribution Module
+
+[TOC]
+
+## `shuffle` - Using Fisher-Yates algorithm to generate a random permutation of a list.
+
+### Status
+
+Experimental
+
+### Description
+
+Applying Fisher-Yates algorithm to generate an unbiased permutation for any list of intrinsic numerical data types.
+
+### Syntax
+
+`result = [[stdlib_stats_distribution_uniform(module):shuffle(interface)]]( list )`
+
+### Arguments
+
+`list`: argument has `intent(in)` and is a rank one array of `integer`, `real`, or `complex` type.
+
+### Return value
+
+Return a randomized rank one array of the input type.
+
+### Example
+
+```fortran
+program demo_shuffle
+    use stdlib_stats_distribution_PRNG, only : random_seed
+    use stdlib_stats_distribution_uniform, only : shuffle
+    implicit none
+    integer :: seed_put, seed_get, i
+    real :: x(10)
+    integer :: n(10)
+    complex :: z(10)
+
+    do i=1, 10
+        n(i) = i
+        x(i) = real(i)
+        z(i) = cmplx(real(i), real(i))
+    end do
+    seed_put = 32165498
+    call random_seed(seed_put, seed_get)    ! set and get current value of seed
+    print *, shuffle(n)                          ! get randomized n
+
+!10   6   9   2   8   1   3   5   7   4
+
+    print *, shuffle(x)                          ! get randomized x
+
+!5.0   10.0   9.0   4.0   3.0   8.0   2.0   1.0   7.0   6.0
+
+    print *, shuffle(z)                          ! get randomized z
+
+!(8.0, 8.0)    (7.0, 7.0)    (4.0, 4.0)    (1.0, 1.0)    (5.0, 5.0)
+!(9.0, 9.0)    (6.0, 6.0)    (3.0, 3.0)    (2.0, 2.0)    (10.0, 10.0)
+
+end program demo_shuffle
+```
+
+## `uniform_distribution_rvs` - uniform distribution random variates
+
+### Status
+
+Experimental
+
+### Description
+
+Without augument the function returns a scalar standard uniformly distributed variate U(0,1) of `real` type with single precision on [0,1].
```

Comment:
Should the default be `single precision` or `double precision`?

Reply:
It is hard to choose. Most people use single precision, some may use double. If most agree on double precision, we can always switch.

Reply:
In `stdlib_stats`, the default is `dp`. However, It might make sense to use `sp` as default here.
Let see what others think about that.

Reply:
There may be merit in making this a subroutine, rather than a function, so that the user can control the output precision (via the input variable)?